rsu-data-controller/src/main/java/com/trihydro/rsudatacontroller/service/RsuService.java

```diff
@@ -106,4 +128,59 @@ private String hexStringToDateTime(String hexString) {
         LocalDateTime date = LocalDateTime.of(year, month, day, hour, minute, 0);
         return date.format(rsuDateTimeFormatter);
     }
+
+    RSU getRSU(String rsuIpv4Address) throws Exception {
+        Connection connection = null;
+		Statement statement = null;
+		ResultSet rs = null;
+        RSU rsu = null;
+        
+
+		try {
+			connection = dbInteractions.getConnectionPool();
+
+			statement = connection.createStatement();
+            
+            // select RSU username, password, and firmware type
+            String selectStatement = "select rc.username, rc.password, sp.nickname FROM rsus " +
+                                "JOIN rsu_credentials AS rc " + //
+                                "ON rsus.snmp_credential_id = rc.credential_id " + //
+                                "JOIN snmp_protocols AS sp " + //
+                                "ON rsus.snmp_protocol_id = sp.snmp_protocol_id " + //
+                                "WHERE rsus.ipv4_address = '" + rsuIpv4Address + "'";
+
+			rs = statement.executeQuery(selectStatement);
+
+            // parse resultSet
+            while (rs.next()) {
+                String rsuUsername = rs.getString("username");
+                String rsuPassword = rs.getString("password");
+                SnmpProtocol snmpProtocol = rs.getString("nickname").equals("NTCIP 1218") ? SnmpProtocol.NTCIP1218 : SnmpProtocol.FOURDOT1;
+                
+                rsu = new RSU(rsuIpv4Address, rsuUsername, rsuPassword, config.getSnmpRetries(), config.getSnmpTimeoutSeconds(), snmpProtocol);
+            }
+
+
+
+		} catch (SQLException e) {
+			e.printStackTrace();
+			return null;
+		} finally {
+			try {
+				// close prepared statement
+				if (statement != null)
+					statement.close();
+				// return connection back to pool
+				if (connection != null)
+					connection.close();
+				// close result set
+				if (rs != null)
+					rs.close();
+			} catch (SQLException e) {
+				e.printStackTrace();
+			}
+		}
```

Comment:
Suggestion (non-blocking): try-with-resources could be used here to simplify the code

Reply:
Updated to use try-with-resources